### ISSUE TITLE
plot predictions shape error with multinomial model

### ISSUE THREAD

**Issue by GStechschulte** [bambinos/bambi#849]
@AlexAndorra raised a question/issue in the LB stats Slack. Posting here.

---

I'm doing something like this:

```python3
bmb.interpret.predictions(
    multinomial_model_bambi,
    idata_multinomial_bambi,
    conditional={
        "cat1": ["A", "B"],
        "name": ["name1", ..., "name10"],
    },
)
```

which works great, but then when trying to get the plot:

```python3
_, ax = bmb.interpret.plot_predictions(
    multinomial_model_bambi,
    idata_multinomial_bambi,
    conditional={
        "cat1": ["A", "B"],
        "name": ["name1", ..., "name10"],
    },
    subplot_kwargs={"main": "cat1", "group": "name", "panel": "name"},
);
```

that raises a weird error: `ValueError: x and y must be the same size`

**Comment by GStechschulte** [comment]
The one thing that I noticed that is not obvious for model's predicting multiple outcomes, e.g multinomial or categorical, is that you will most likely want to plot each outcome as a hue or panel (facet). But if you do not know a priori that Bambi adds the column `estimate_dim`, then you will not pass it as a subplot kwarg.

**Comment by tomicapretto** [comment]
I'm not sure I understand if this is a bug or something we should document better :sweat_smile:

**Comment by GStechschulte** [comment]
It's definitely something I would like to document better. From an interface point of view, I think the first plot in my comment above should render because: (1) the `interpret.predictions` successfully builds the data frame, and (2) none of `ValueErrors` for `.interpret.plot_predictions` are raised. 

I will work on this :)

**Comment by tomicapretto** [comment]
@GStechschulte amazing, just ping me if you want review or help :)

**Comment by GStechschulte** [comment]
Long overdue, but here are my thoughts:

When dealing with models that can predict multiple outcomes, we add a column `estimate_dim` in the summary dataframe where each value represents an outcome. The problem is that the plotting functions don't detect and handle this column when `subplot_kwargs` are not explicitly provided causing the 

`ValueError: x and y must be the same size`

I think plotting when a user has a multinomial model should "just work", i.e. plotting functions should detect that they will be plotting data that has multiple outcomes. An alternative could be to raise a user warning (or error?) along the lines of

`Detected multiple outcomes. Consider passing estimate_dim to subplot_kwargs.`

Nonetheless, we should also have better documentation on this.

**Comment by tomicapretto** [comment]
> I think plotting when a user has a multinomial model should "just work

I agree with this. Probably it can be determined from the model's metadata? I don't remember it exactly right now, but we must have some attribute that tells whether the response is multivariate or not.

**Comment by GStechschulte** [comment]
I would think so as well. I will look into this.

**Comment by GStechschulte** [comment]
The interpret refactor resolved this via the following: if the model has a likelihood where the model predicts multiple outcomes, then the summary dataframe has an `estimate_dim` column. The plotting code detects if an `estimate_dim` column exists and will add this column to the subplot configuration, i.e., `main`, `group`, or `panel`.
